packages/ilios-common/addon/components/mesh-manager.hbs

```diff
@@ -1,139 +1,131 @@
-    {{#if @targetItemTitle}}
-      <h2 class="target-title">
-        {{t "general.selectMeshFor" title=@targetItemTitle}}
-      </h2>
-    {{/if}}
-    {{#if this.sortedTerms.length}}
-      <ul class="selected-terms">
-        {{#each this.sortedTerms as |term|}}
-          <li>
-            {{#if @editable}}
-              <button
-                type="button"
-                {{on "click" (fn @remove term)}}
-              >
-                <span class="term-title">
-                  {{term.name}}
-                </span>
-                <span class="term-details">
-                  {{term.id}}
-                  {{#if term.deleted}}
-                    - <span class="deprecated">({{t "general.deprecatedAbbreviation"}})</span>
-                  {{else if term.trees.length}}
-                    - <MeshDescriptorLastTreeNumber @descriptor={{term}} />
-                  {{/if}}
-                </span>
-                <FaIcon @icon="xmark" class="remove" />
-              </button>
-            {{else}}
+<section class="mesh-manager" data-test-mesh-manager>
+  {{#if @targetItemTitle}}
+    <h2 class="target-title">
+      {{t "general.selectMeshFor" title=@targetItemTitle}}
+    </h2>
+  {{/if}}
+  {{#if this.sortedTerms.length}}
+    <ul class="selected-terms">
+      {{#each this.sortedTerms as |term|}}
+        <li>
+          {{#if @editable}}
+            <button type="button" {{on "click" (fn @remove term)}}>
               <span class="term-title">
                 {{term.name}}
               </span>
               <span class="term-details">
-              {{term.id}}
+                {{term.id}}
                 {{#if term.deleted}}
-                  - <span class="deprecated">({{t "general.deprecatedAbbreviation"}})</span>
+                  -
+                  <span class="deprecated">({{t "general.deprecatedAbbreviation"}})</span>
                 {{else if term.trees.length}}
-                  - <MeshDescriptorLastTreeNumber @descriptor={{term}} />
+                  -
+                  <MeshDescriptorLastTreeNumber @descriptor={{term}} />
+                {{/if}}
+              </span>
+              <FaIcon @icon="xmark" class="remove" />
+            </button>
+          {{else}}
+            <span class="term-title">
+              {{term.name}}
+            </span>
+            <span class="term-details">
+              {{term.id}}
+              {{#if term.deleted}}
+                -
+                <span class="deprecated">({{t "general.deprecatedAbbreviation"}})</span>
+              {{else if term.trees.length}}
+                -
+                <MeshDescriptorLastTreeNumber @descriptor={{term}} />
+              {{/if}}
+            </span>
+          {{/if}}
+        </li>
+      {{/each}}
+    </ul>
+  {{/if}}
+  {{#if @editable}}
+    <div class="mesh-search" data-test-mesh-search>
+      <label for={{this.meshSearchInputId}}>{{t "general.searchMesh"}}:</label>
```

Comment:
this change has become unnecessary again, you can revert to the previous form of using the `unique-id as |templateId|` approach.